### ISSUE TITLE
Script extension mismatch in jupyter_download_role

### ISSUE THREAD

**Issue by renefritze** [jupyter/jupyter-sphinx#133]
I'm using `juptyer-sphinx` in document with first a bash kernel and then a python kernel (no ids used). This produces 4 files in my `build/jupyter_execute` subdir:
```
tutorial_external_solver_1.ipynb  tutorial_external_solver_1.py  
tutorial_external_solver.ipynb  tutorial_external_solver.sh
```

Now the download role always appends ".py" to the script. Same if I set and use a id for the bash kernel. So to me it seems that there's currently no way for me to offer the bash script for download. At least using the jupyter-download role, manually could be a work around I guess?

**Comment by akhmerov** [comment]
Indeed, right now this isn't supported properly. The logic for determining the link target is in [this line](https://github.com/jupyter/jupyter-sphinx/blob/b80c8497bff09fd2bf718aaad171f9052997ed24/jupyter_sphinx/ast.py#L402), while the file extension is correctly set by the notebook that is executed.

@chrisjsewell since you've just reviewed that code, what would it take to correctly fix the file extensions for the download roles? A post-transform?

**Comment by chrisjsewell** [comment]
ah yes I noticed, but willfully ignored it in #132 lol (to not complicate things further), and was planning to bring it up in a separate issue/PR. There didn't look to be a trivial solution, but I'll try to have a look in the next day or 2 and I'm sure we can figure something out
